main.py

```diff
@@ -11,40 +11,28 @@
 This module is built on top of the Pydle system.
 
 """
+import logging
+from sys import argv
+
 from pydle import ClientPool, Client
+
+import config
+
+# this feels really ugly, im probably going to replace this with argparse or similar.
+if len(argv) >= 2:  # assume first argument is the config file to use
+    config_filepath = argv[1]
+else:  # we didn't get one
+    config_filepath = "./config/config.template.json"
```

Comment:
I may be wrong, but this probably could be done in `config.py` itself, which would get rid of doing this before the imports (though the logging stuff will have to remain here) and also make it certain that the config has been read when it is imported and used.

I also agree that this should be using argparse, which is a great bit of standard library. This would make it much more expandable in case we want to add more command-line arguments in the future. Argparse also automatically provides a lovely `-h` for printing info on available arguments.

Reply:
Wouldn't have to pass argv explicitly, can just access sys.argv from config.py